### PR TITLE
Add exponent button to operator list and add unit tests.

### PR DESCRIPTION
**Pull request by codebaum**

### CODE REVIEW

File: app/src/main/java/com/mobiquity/androidunittests/ui/activity/CalculatorActivity.java [hunk 1]
```diff
@@ -67,12 +68,12 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
         ButterKnife.bind(this);
 
         numericPad.addOnNumberClickedListener(number -> presenter.handleCalculatorButtonPress(Integer.toString(number)));
-        ButterKnife.apply(calculatorButtons, (button, index) -> {
+        ButterKnife.apply(calculatorButtons, (ButterKnife.Action<? super TextView>) (button, index) -> {
             String operator = button.getText().toString();
```
Comment (jlw8ke) [inline app/src/main/java/com/mobiquity/androidunittests/ui/activity/CalculatorActivity.java]: I believe these casts are redundant.  You should be able to call this method without having to cast.

Comment (jlw8ke) [inline app/src/main/java/com/mobiquity/androidunittests/ui/activity/CalculatorActivity.java]: I believe this cast is redundant.  You should be able to call this method without needing to cast.

File: app/src/main/java/com/mobiquity/androidunittests/ui/view/NumericPad.java [hunk 2]
```diff
@@ -57,7 +57,7 @@ public void removeOnNumberClickedListener(OnNumberClickedListener listener) {
     private void initView() {
         inflate(getContext(), R.layout.numeric_pad, this);
         ButterKnife.bind(this, this);
-        ButterKnife.apply(numericButtons, (button, index) -> {
+        ButterKnife.apply(numericButtons, (ButterKnife.Action<? super Button>) (button, index) -> {
             button.setText(String.valueOf(index));
```
Comment (jlw8ke) [inline app/src/main/java/com/mobiquity/androidunittests/ui/view/NumericPad.java]: redundant cast

File: app/src/test/java/com/mobiquity/androidunittests/calculator/CalculatorTest.java [hunk 3]
```diff
@@ -105,6 +106,29 @@ public void testEvaluate_AdditionMultiply() throws Exception {
         assertThat(result).isWithin(expectedResult);
     }
 
+    @Test
```
Comment (jlw8ke) [inline app/src/test/java/com/mobiquity/androidunittests/calculator/CalculatorTest.java]: Nice test c:

Can you  change the comments in the test to match what you're testing?

File: app/src/test/java/com/mobiquity/androidunittests/calculator/InfixInputParserTest.java [hunk 4]
```diff
@@ -144,7 +146,22 @@ public void testToPostfix_SimpleMultiply() {
                 .inOrder();
     }
```
Comment (jlw8ke) [inline app/src/test/java/com/mobiquity/androidunittests/calculator/InfixInputParserTest.java]: update the comment to match the test

File: app/src/test/java/com/mobiquity/androidunittests/calculator/input/operator/ExponentOperatorTest.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        assertThat(exponentOperator.getPrecedence())
+                .isEqualTo(Operator.Precedence.POWER_PRECENDENCE.getValue());
+    }
+
+    @Test
+    public void testShouldBeLeftAssociative() {
+        assertThat(exponentOperator.isLeftAssociative()).isTrue();
+    }
+
+    @Test
```
Comment (jlw8ke) [inline app/src/test/java/com/mobiquity/androidunittests/calculator/input/operator/ExponentOperatorTest.java]: update the name of the test for exponents

File: app/src/test/java/com/mobiquity/androidunittests/calculator/CalculatorTest.java [hunk 6]
```diff
@@ -105,6 +106,29 @@ public void testEvaluate_AdditionMultiply() throws Exception {
...
+    public void testEvaluate_Exponent() throws Exception {
+        //Test: 1 + 2 * 3 = 7
```
Comment (jlw8ke) [inline app/src/test/java/com/mobiquity/androidunittests/calculator/CalculatorTest.java]: Can you change the comments in the test to match what you're testing? It also might be worth having a test with an exponent and another operation to test order of operations.

File: app/src/test/java/com/mobiquity/androidunittests/calculator/input/operator/ExponentOperatorTest.java [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    public void testExecute_ShouldMultiplyOperands() {
```
Comment (jlw8ke) [inline app/src/test/java/com/mobiquity/androidunittests/calculator/input/operator/ExponentOperatorTest.java]: update the name of the test for exponents

File: app/src/main/java/com/mobiquity/androidunittests/ui/activity/CalculatorActivity.java (hunk 1 above)
Comment (codebaum) [inline app/src/main/java/com/mobiquity/androidunittests/ui/activity/CalculatorActivity.java]: Android Studio was showing a red error for me (although I think the app was still runnable somehow), so I just did this to get it to stop complaining.

File: app/src/main/java/com/mobiquity/androidunittests/ui/view/NumericPad.java (hunk 2 above)
Comment (codebaum) [inline app/src/main/java/com/mobiquity/androidunittests/ui/view/NumericPad.java]: see above response

File: app/src/main/java/com/mobiquity/androidunittests/ui/presenter/CalculatorPresenter.java [hunk 8]
```diff
@@ -76,7 +78,11 @@ private void evaluate(boolean isResult) {
     private String getDisplayString() {
         String displayString = "";
         for(String item : expression) {
-            displayString += item;
+            if (item.equals("^")) {
```
Comment (codebaum) [inline app/src/main/java/com/mobiquity/androidunittests/ui/presenter/CalculatorPresenter.java]: Will do... I had some other logic at first in there that I got rid of.

File: app/src/test/java/com/mobiquity/androidunittests/calculator/CalculatorTest.java (hunk 6 above)
Comment (codebaum) [inline app/src/test/java/com/mobiquity/androidunittests/calculator/CalculatorTest.java]: will change the comments

File: app/src/test/java/com/mobiquity/androidunittests/calculator/input/operator/ExponentOperatorTest.java (hunk 7 above)
Comment (codebaum) [inline app/src/test/java/com/mobiquity/androidunittests/calculator/input/operator/ExponentOperatorTest.java]: will do

File: app/src/test/java/com/mobiquity/androidunittests/calculator/InfixInputParserTest.java (hunk 4 above)
Comment (codebaum) [inline app/src/test/java/com/mobiquity/androidunittests/calculator/InfixInputParserTest.java]: will do

**Comment by codebaum** [comment]
@jlw8ke updated

**Pull request merged**